metadata_backend/api/handlers.py

```diff
@@ -303,8 +351,24 @@ async def get_folder(self, req: Request) -> Response:
         :param req: GET request
         :returns: JSON response containing object folder
         """
-        # folder_id = req.match_info['folderId']
-        raise web.HTTPNotImplemented
+        folder_id = req.match_info['folderId']
+        db_client = req.app['db_client']
+        db_service = DBService("folders", db_client)
+        try:
+            folder = await db_service.read("folder", folder_id)
+            if not folder:
+                reason = f"Folder with id {folder_id} not found."
+                LOG.error(reason)
+                raise web.HTTPNotFound(reason=reason)
+        except (ConnectionFailure, OperationFailure) as error:
+            reason = f"Error happened while getting folder: {error}"
+            LOG.error(reason)
+            raise web.HTTPBadRequest(reason=reason)
+
+        folder.pop('_id', None)  # remove unneccessary mongodb id from result
```

Comment:
btw you can just do `del doc["_id"]` here, since you don't use the value returned by `.pop()`.

Reply:
`del` will raise `KeyError` if it does not find the key, I think this is ok. Although it might never be raised it is better to avoid and fail silently as it is not a relevant key.

Reply:
well, but mongoDb will always return _id, right? if you think failing silently is better, this should be also fixed: https://github.com/CSCfi/metadata-submitter/blob/2de12d9f26837025a172eb1d60ea5d42d713351a/metadata_backend/api/operators.py#L497

Reply:
maybe we can prevent it on query level, i think we can merge as is.